idling/v1alpha2/types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NB: slice types *must* be omitempty, otherwise the CRD
+// validator will complain that slices can't be null (since nil slices
+// become null, not empty arrays).
+
+// IdlerSpec defines the desired state of Idler
+type IdlerSpec struct {
+	// WantIdle represents the desired state of idling
+	WantIdle bool `json:"wantIdle"`
+	// TargetScalables contains the collection of scalables that
+	// are idled/unidled together.
+	TargetScalables []CrossGroupObjectReference `json:"targetScalables,omitempty"`
+	// TriggerServiceNames contains the collection of services that shold
+	// trigger unidling.  Their corresponding endpoints objects will be
+	// used to determine whether or not unidling is successful.
+	TriggerServiceNames []string `json:"triggerServiceNames,omitempty"`
```

Comment:
Are the services here supposed to select the resources above?  What keeps them in sync?  Are they allowed to be different?  What if the list of targetScalables in incomplete?

Reply:
they're allowed to be *somewhat* different.  Nothing explicitly keeps them in sync.  The invariant that needs to be preserved, which is referred to, but I could mention more specifically, is that scaling up all the scalables needs to cause each trigger service to have at least one endpoint (more on this below).